src/wp-includes/post.php

```diff
@@ -7489,6 +7489,22 @@ function update_post_author_caches( $posts ) {
 	cache_users( $author_ids );
 }
 
+/**
+ * Prime post parent post caches.
+ *
+ * @since 6.1.0
+ *
+ * @param WP_Post[] $posts Array of Post objects
+ */
+function update_post_parent_caches( $posts ) {
+	$parent_ids = wp_list_pluck( $posts, 'post_parent' );
```

Comment:
Should we consider an array of post IDs as an additional possible value for the `$posts` parameter?
```suggestion
	$posts = array_map( 'get_post', $posts );
	$parent_ids = wp_list_pluck( $posts, 'post_parent' );
```

Reply:
The type is hinted in the docblock. There is no reason in core to support this.

Reply:
You can also just do this myself. 

```
$posts = array_map( 'get_post', $post_ids );
update_post_parent_caches( $posts );
```